lib/iris/analysis/maths.py

```diff
@@ -569,11 +577,11 @@ def _binary_op_common(operation_function, operation_noun, cube, other,
         other = _broadcast_cube_coord_data(cube, other, operation_noun, dim)
     elif isinstance(other, iris.cube.Cube):
         try:
-            np.broadcast_arrays(cube.data, other.data)
+            BA.broadcast_arrays(cube._my_data, other._my_data)
```

Comment:
After looking at `BroadcastArray.broadcast_arrays` it appears that all of the work done involves calculations based on the shape of the arrays so I'm not sure there's any benefit to turning already loaded data back into a lazy array here.